src/Maker/MakeTest.php

```diff
@@ -0,0 +1,209 @@
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Exception\RuntimeCommandException;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputAwareMakerInterface;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Component\BrowserKit\History;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\CssSelector\CssSelectorConverter;
+use Symfony\Component\Panther\PantherTestCaseTrait;
+
+/**
+ * MakeTest.
+ *
+ * @author Kévin Dunglas <kevin@dunglas.fr>
+ * @author Javier Eguiluz <javier.eguiluz@gmail.com>
+ * @author Ryan Weaver <weaverryan@gmail.com>
+ */
+final class MakeTest extends AbstractMaker implements InputAwareMakerInterface
+{
+    public static function getCommandName(): string
+    {
+        return 'make:test';
+    }
+
+    // TODO: remove this method when removing make:unit-test and make:functional-test
+    public static function getCommandAliases(): iterable
+    {
+        yield 'make:unit-test';
+        yield 'make:functional-test';
+    }
+
+    public static function getCommandDescription(): string
+    {
+        return 'Creates a new test class';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConf)
+    {
+        $command
+            ->addArgument('name', InputArgument::OPTIONAL, 'The name of the test class (e.g. <fg=yellow>BlogPostTest</>)')
+            ->addArgument('type', InputArgument::OPTIONAL, 'The type of test: <fg=yellow>unit</> (PHPUnit), <fg=yellow>integration</> (WebTestCase) or <fg=yellow>e2e</> (PantherTestCase)')
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeTest.txt'));
+
+        $inputConf->setArgumentAsNonInteractive('type');
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
+    {
+        // TODO: remove the following block when removing make:unit-test and make:functional-test
+        $currentCommand = $input->getFirstArgument();
+        switch ($currentCommand) {
+            case 'make:unit-test':
+                $input->setArgument('type', 'unit');
+                $io->caution('The "make:unit-test" command is deprecated, use "make:test" instead.');
+
+                return;
+
+            case 'make:functional-test':
+                $input->setArgument('type', trait_exists(PantherTestCaseTrait::class) ? 'e2e' : 'integration');
+                $io->caution('The "make:functional-test" command is deprecated, use "make:test" instead.');
+
+                return;
+        }
+
+        if (null !== $input->getArgument('type')) {
+            $this->normalizeType($input);
+
+            return;
+        }
+
+        $input->setArgument(
+            'type',
+            $io->choice($command->getDefinition()->getArgument('type')->getDescription(), ['unit', 'integration', 'e2e'])
```

Comment:
What about not re-using the `type` argument definition? I'd rather say something like this (which probably includes printing some thins before the question) - I'm actually re-using your suggestion from another comment, which I really like:

> Which test type would you like?
> 
> 1) TestCase: basic PHPUnit test
> 2) KernelTestCase: basic tests that have access to Symfony services
> 3) WebTestCase: to run browser-like scenarios, but that don't execute JavaScript code
> 4) ApiTestCase: to run API-oriented scenarios (needs API Platform)
> 5) PantherTestCase: to run e2e scenarios, using aa real-browser or HTTP client and a real web server